backend/thirdparty/provider_microsoft.go

```diff
@@ -0,0 +1,217 @@
+				Email: address.Address,
+				// Assume it is verified because it looks like UPN suffixes cannot be set to unverified domains.
+				Verified: true,
+				Primary:  true,
+			}
+		}
+	} else {
+		if idTokenClaims.Email != "" {
+			if idTokenClaims.IsEmailVerified() {
+				email = &Email{
+					Email:    idTokenClaims.Email,
+					Verified: true,
+					Primary:  true,
+				}
+			} else {
+				email = &Email{
+					Email:    idTokenClaims.Email,
+					Verified: false,
+					Primary:  true,
+				}
+			}
+		}
+	}
+
+	if email == nil {
+		return nil, errors.New("unable to find email with Microsoft provider")
+	}
+
+	data := &UserData{}
+	data.Emails = append(data.Emails, *email)
+
+	data.Metadata = &Claims{
+		Issuer:            parsedIDToken.Issuer(),
+		Subject:           parsedIDToken.Subject(),
+		Name:              idTokenClaims.Name,
+		PreferredUsername: idTokenClaims.PreferredUsername,
+		Email:             email.Email,
+		EmailVerified:     email.Verified,
+	}
+
+	return data, nil
+}
+
+func (p microsoftProvider) Name() string {
+	return "microsoft"
+}
+
+func (p microsoftProvider) issuerValidator() jwt.ValidatorFunc {
+	var microsoftIssuerRegexp = regexp.MustCompile("^https://login[.]microsoftonline[.]com/([^/]+)/v2[.]0/?$")
+	validator := jwt.ValidatorFunc(func(_ context.Context, t jwt.Token) jwt.ValidationError {
+		if !microsoftIssuerRegexp.MatchString(t.Issuer()) {
+			return jwt.NewValidationError(fmt.Errorf(`%s is not a valid microsoft issuer`, t.Issuer()))
+		}
+		return nil
+	})
+	return validator
+}
+
+type microsoftIdTokenClaims struct {
+	Email                              string `mapstructure:"email"`
+	Name                               string `mapstructure:"name"`
+	PreferredUsername                  string `mapstructure:"preferred_username"`
+	UserPrincipalName                  string `mapstructure:"upn"`
+	XMicrosoftEmailDomainOwnerVerified any    `mapstructure:"xms_edov"`
+}
+
+// IsEmailVerified checks if the email used is verified. Shameless plug from Supabase's GoTrue fork
+// See: https://github.com/supabase/gotrue/blob/master/internal/api/provider/oidc.go#L221
+// See also: https://www.descope.com/blog/post/noauth
+func (c *microsoftIdTokenClaims) IsEmailVerified() bool {
+	emailVerified := false
+
+	edov := c.XMicrosoftEmailDomainOwnerVerified
+
+	// If xms_edov is not set, and an email is present or xms_edov is true,
+	// only then is the email regarded as verified.
+	// https://learn.microsoft.com/en-us/azure/active-directory/develop/migrate-off-email-claim-authorization#using-the-xms_edov-optional-claim-to-determine-email-verification-status-and-migrate-users
+	if edov == nil {
+		// An email is provided, but xms_edov is not -- probably not
+		// configured, so we must assume the email is verified as Azure
+		// will only send out a potentially unverified email address in
+		// single-tenant apps (which we do not support - only the multi-tenant
+		// + public account type).
+		emailVerified = c.Email != ""
```

Comment:
I would sanity check if the email is not filed with spaces or a non email address. But thats my opinion from a zero trust in 3rd party applications standpoint. @FlxMgdnz @FreddyDevelop what is your opinion on that one?